lang/csharp/src/apache/main/Util/LogicalUnixEpochType.cs

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+
+namespace Avro.Util
+{
+    /// <summary>
+    /// Base for all logical type implementations that are based on the Unix Epoch date/time.
+    /// </summary>
+    public abstract class LogicalUnixEpochType<T> : LogicalType
+        where T : struct
+    {
+        /// <summary>
+        /// The date and time of the Unix Epoch.
+        /// </summary>
+        protected static readonly DateTime UnixEpochDateTime = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
```

Comment:
I think this should be `DateTimeKind.Unspecified`. 

The spec says:

> The timestamp-millis logical type represents an instant on the global timeline, independent of a particular time zone or calendar,

by specifying a kind of UTC, the returned values are explicitly not independent of a particular timezone.

the spec does go on to say:

> where the long stores the number of milliseconds from the unix epoch, 1 January 1970 00:00:00.000 UTC

(i.e. includes time zone in the epoch), but I don't think that is intended or what we want.